### PR TITLE
Split IGraphicsDevice into IPlatformWindow and IGraphicsContext

### PR DESCRIPTION
**Pull request by pchote**
This PR implements the first three steps from #15211.  The changes here are limited to moving or renaming code, there are no logic changes.  If you want to see how the last step builds on the changes here you can take a look at my graphics-device branch, which I will PR after this one is merged.

The questions to ask yourself while reviewing are:
* Does the new ownership model make sense looking forward?
* Do the new interface and class names make sense?
* Have I moved something that I shouldn't, or forgotten something that I should?

### CODE REVIEW

**Comment by pchote** [comment]
My comment about no logic changes isn't completely true, as I of course had to change the constructors and Dispose methods when splitting `Sdl2GraphicsDevice` &rarr; `Sdl2PlatformWindow` + `Sdl2GraphicsContext`.

File: OpenRA.Platforms.Default/Sdl2GraphicsContext.cs [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+				return;
+
+			disposed = true;
+			if (context != IntPtr.Zero)
+			{
+				SDL.SDL_GL_DeleteContext(context);
+				context = IntPtr.Zero;
+			}
+
+			SDL.SDL_Quit();
```
Comment (RoosterDragon) [inline OpenRA.Platforms.Default/Sdl2GraphicsContext.cs]: https://github.com/OpenRA/OpenRA/issues/15188 suggests `SDL.SDL_Quit` needs to live in the window layer, not GL.

Comment (pchote) [reply OpenRA.Platforms.Default/Sdl2GraphicsContext.cs]: Good catch. I forgot to delete this when copying and editing the dispose method - it was actually defined in both places.

**Review by reaperrr — APPROVED** [review]

**Review by RoosterDragon — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/09c135972c48564a8a97308f3b4729ec52a84e09 (note that I moved the renderer thread PRs to their own entry)

**Pull request merged**